app/controllers/project2.php

```diff
@@ -0,0 +1,279 @@
+		$inProjects =  $user->inProjects()->orderBy('created_at', 'desc')->take(5)->get();
+		
+		return View::make('projects.index', compact(['projects','inProjects','counter','pTitle']));
+	}
+
+	/**
+	 * Create the new project
+	 *
+	 * @return Response
+	 */
+	public function create()
+	{	
+		// Rules
+		$rules	= array('name' => 'required',);
+
+		// Create validation 
+		$validator = Validator::make( Input::all(), $rules);
+
+		if ( $validator->fails() ) {
+			return Redirect::back()->withErrors($validator)->withInput();
+		}
+		
+		$project 			=	new Project;
+		$project->name 		=	Input::get('name');
+		$project->client_id =	Input::get('client_id');
+		$project->user_id	=	Auth::id();
+		$project->save();
+
+		return Redirect::back()->with('success', Input::get('name') ." has been created.");
+	}
+
+	/**
+	 * Display the specified resource.
+	 * GET /projects/{id}
+	 *
+	 * @param  int  $id
+	 * @return Response
+	 */
+	public function show($id)
+	{   	
+		$project 		=	Project::find($id);
+
+		// Must be refactored as a filter
+		if ( $project->isOwner() == false && $project->isMember() == false ) {
+			return Redirect::to('/hud');
+		}
+
+		$tasks 			=	$project->tasks()->where('state','incomplete')->orderBy("updated_at", "desc")->get();
+		$completedTasks	=	$project->tasks()->where('state','complete')->get();
+		$taskCount 		=	count($tasks);
+		$completedCount =	count($completedTasks);
+		$total_weight	=	$project->tasks()->where('state','incomplete')->sum('weight');
+		$credentials   	=	$project->credentials;
+		$owner_id		=	$project->user_id;
+		$members 		= 	$project->members()->get();
+
+		$pTitle 		=	$project->name; 
+			
+		return  View::make('projects.show', compact(
+											[
+												'owner_id',
+												'project',
+												'tasks',
+												'completedTasks',
+												'taskCount',
+												'total_weight',
+												'completedCount',
+												'members',
+												'pTitle'
+											]));
+	}
+
+	/**
+	 * Show the form for editing the specified resource.
+	 * GET /projects/{id}/edit
+	 *
+	 * @param  int  $id
+	 * @return Response
+	 */
+	public function edit($id)
+	{
+		$project 	= 	Project::find($id);
+
+		if($project->isOwner() == false){
+			return Redirect::to('/');
+		}
+
+		$pTitle 		=	"Edit " . $project->name;
+		$total_weight	=	$project->tasks()->where('state','incomplete')->sum('weight');
+		$owner_id		=	$project->user_id;
+		$members 		= 	$project->members()->get();
+
+		return View::make('projects.edit', compact(['project','pTitle','owner_id','total_weight','members']));
+	}
+
+	/**
+	 * Update the specified resource in storage.
+	 * PUT /projects/{id}
+	 *
+	 * @param  int  $id
+	 * @return Response
+	 */
+	public function update($id)
+	{
+		$project 	= Project::find($id);
+
+       // Validation
+        $validator = Validator::make(
+            array(
+            	'name' 				=>	Input::get("name"),
+            	'github'			=>	Input::get("github"),
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Define a constant instead of duplicating this literal "github" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://172.27.59.137:9000/coding_rules#rule_key=php%3AS1192)